### ISSUE TITLE
pythonPackages.rsa is broken in recent channels/unstable

### ISSUE THREAD

**Issue by akamaus** [NixOS/nixpkgs#53116]
## Issue description
I can no longer build pythonPackages.rsa on nixos with nixpkgs fetched from channels/nixos-unstable (201d739b0ffbeb)

```
% nix-shell-unstable   -p python36Packages.rsa                                                                                        
these derivations will be built:
  /nix/store/mcncvdwadrq1335bwy279dg7q0sxhvhp-python3.6-rsa-4.0.drv
building '/nix/store/mcncvdwadrq1335bwy279dg7q0sxhvhp-python3.6-rsa-4.0.drv'...
unpacking sources
unpacking source archive /nix/store/km959nhndbrasamr0zj49w4hswiss72z-rsa-4.0.tar.gz
source root is rsa-4.0
setting SOURCE_DATE_EPOCH to timestamp 1537098221 of file rsa-4.0/setup.cfg
patching sources
configuring
building
Traceback (most recent call last):
  File "nix_run_setup", line 8, in <module>
    exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\\r\\n', '\\n'), __file__, 'exec'))
  File "setup.py", line 20, in <module>
    long_description = f.read()
  File "/nix/store/qbyckcddc74s2fi060n8xqcn8gygrhjl-python3-3.6.8/lib/python3.6/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 898: ordinal not in range(128)
builder for '/nix/store/mcncvdwadrq1335bwy279dg7q0sxhvhp-python3.6-rsa-4.0.drv' failed with exit code 1
```

Interestingly, it builds successfully on another host with userspace nix installation.

The following patch fixes the problem for me, but I'm unsure about the root cause.

```
diff --git a/pkgs/development/python-modules/rsa/default.nix b/pkgs/development/python-modules/rsa/default.nix
index c062aad4c4b..ed47ac31c75 100644
--- a/pkgs/development/python-modules/rsa/default.nix
+++ b/pkgs/development/python-modules/rsa/default.nix
@@ -18,6 +18,10 @@ buildPythonPackage rec {
   checkInputs = [ unittest2 mock ];
   propagatedBuildInputs = [ pyasn1 ];
 
+  preConfigure = ''
+       substituteInPlace setup.py --replace "open('README.md')" "open('README.md',encoding='utf-8')"
+  '';  
+
   meta = with stdenv.lib; {
     homepage = https://stuvel.eu/rsa;
     license = licenses.asl20;
```

**Comment by FRidh** [comment]
https://discourse.nixos.org/t/python36packages-python-jose-fails-with-ascii-codec-error-which-breaks-jupyter/1786

**Comment by FRidh** [comment]
Pushed d65a3efc3f680f6229883778f1e77c1ae3f15709

**Comment by puckipedia** [comment]
This does not seem to have fixed it for me, running on master. I had to apply this patch to `rsa` as well.

**Comment by FRidh** [comment]
Silly me, I applied the changeset to the wrong derivation. Should be fixed as of 4a7c9e5af23ca8c0116c73ff89583f7be8e01dcf
